### DISCUSSION TITLE
Connector column does not call cloud function properly

### DISCUSSION THREAD

**Discussion by pierpo** [AntlerVC/firetable#1357] [Support / Q&A] [answered]
Hey!

I am trying to set up a Connector column. I used all the default values.
![image](https://github.com/rowyio/rowy/assets/6313316/ab631852-6034-4768-b549-8cca365e5ea1)

When I set it up, I get an infinite loader.
![image](https://github.com/rowyio/rowy/assets/6313316/249d6a67-be1b-4d9c-b272-ce1d44599efb)

By inspecting, I see that I get a 404 on the cloud function call.
![image](https://github.com/rowyio/rowy/assets/6313316/e8ac35ba-b3e3-4d92-acd7-f79ac3702953)

I do have rowy run configured (even though the version is a little old)
![image](https://github.com/rowyio/rowy/assets/6313316/a390725c-26e2-4cb9-bb1d-eb814c224176)

Any idea?

Thanks 😊

**Comment by BeeBombshell** [comment]
Hey @pierpo, we previously had an update that requires some modifications in the scripts. Try using the following instead:

```js
const connector: Connector = async ({ query, row, user, logging }) => {
  logging.log("connector started");
  const items = [
    {
      id:"a",
      word:"apple",
      emoji:"🍎"
    },
    {
      id:"b",
      word:"banana",
      emoji:"🍌"
    },
    {
      id:"c",
      word:"cherry",
      emoji:"🍒"
    },
    {
      id:"d",
      word:"duck",
      emoji:"🦆"
    },
    {
      id:"e",
      word:"eggplant",
      emoji:"🍆"
    },
];
  return items.filter(i=> i.word.toLowerCase().startsWith(query.toLowerCase()))
};

export default connector;
```
Check the connector configuration in the screenshot below:
<img width="1790" alt="Screenshot 2023-08-09 at 12 22 13 PM" src="https://github.com/rowyio/rowy/assets/53828745/c1ee440f-b7a1-41a5-a31c-654eb51984b6">

You can also update your outdated Rowy Run version to keep up with the new updates and fixes.

**Reply by pierpo** [reply]
Hey, thank you for the answer! I tried updating my Rowy Run, and also tried to replicate what you gave me. I still have a 4xx error. This time, this is a 401.

If I check the logs of the connector, I have none. It also feels like the cloud function was not deployed at all, since I don't see any new function in my GCP cloud functions.

Is there a way to understand what's going on underneath?

The settings I put:
<img width="1031" alt="image" src="https://github.com/rowyio/rowy/assets/6313316/fc07452f-7d33-4de4-b056-cc21d97487d1">

No new deployments:
<img width="640" alt="image" src="https://github.com/rowyio/rowy/assets/6313316/8bceed9c-d033-4192-850e-7d0f1c078674">

And I still get my infinite loader 
<img width="208" alt="image" src="https://github.com/rowyio/rowy/assets/6313316/a8124f00-324c-4bbd-a103-7dfe502c1efc">

**Reply by pierpo** [reply, marked as answer]
OK, it turns out that our rowy run was still outdated. We saw that the last deploy on GCP was quite old, so we ran the commands to manually setup Rowy Run.

From the docs:
```bash
gcloud run deploy rowy-backend --image gcr.io/rowy-run/rowy-backend --platform managed --memory 1Gi --allow-unauthenticated --service-account rowy-backend@YOUR_PROJECT_ID.iam.gserviceaccount.com
gcloud run deploy rowy-hooks --image gcr.io/rowy-run/rowy-hooks --platform managed --memory 256Mi --allow-unauthenticated --service-account rowy-hooks@YOUR_PROJECT_ID.iam.gserviceaccount.com
```

<img width="956" alt="image" src="https://github.com/rowyio/rowy/assets/6313316/ae727e61-aa77-4f19-95be-d05e84cfca99">
